### PR TITLE
Fix Firefox certificate warnings

### LINKED ISSUE #3398: HTTPS: Firefox does not trust parcel-generated certificates after correct installation
# 🐛 bug report

Despite parcel-generated certificate working in Chrome, Firefox is not trusting the certificate with the error `MOZILLA_PKIX_ERROR_CA_CERT_USED_AS_END_ENTITY`.

## 🤔 Expected Behavior

Certificate should be trusted by Firefox.

## 😯 Current Behavior

Certificate is not trusted by Firefox.

## 💁 Possible Solution

According to the bottom of [this page](https://wiki.mozilla.org/SecurityEngineering/x509Certs) it is due to a feature enabled [here](https://github.com/parcel-bundler/parcel/blob/2c1499596680be6daf158334f4146b80592ed4be/packages/core/parcel-bundler/src/utils/generateCertificate.js#L66):

"A certificate with a basic constraints extension with cA:TRUE is being used as an end-entity certificate"

Removing this feaure and generating a new certificate works as expected in both Firefox and Chrome ([details here](https://github.com/shanehandley/parcel-cert-gen-firefox)).

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 1.12.3
| Node             | v8.12.0
| npm/Yarn         | npm 6.10.3
| Operating System | MacOS Mojave 10.14.6

### PR DESCRIPTION
**Pull request by DeMoorJasper**
# ↪️ Pull Request

According to #3398 firefox throws errors on our self-signed certificates, this is due to a misconfiguration in the generate script we use.

This PR fixes the script by removing the feature that says our certificates are issued by a CA.

Closes #3398

## 🚨 Test instructions

Run dev server in firefox

## ✔️ PR Todo

- [x] Added/updated unit tests for this change
- [x] Filled out test instructions (In case there aren't any unit tests)
- [x] Included links to related issues/PRs

### CODE REVIEW

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>packages/benchmarks/kitchen-sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 10.43s | +50.00ms |
| Cached | 8.64s | -104.00ms |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.ff215454.webp | 102.94kb | +0.00b | 167.00ms | +17.00ms ⚠️ |
| dist/modern/parcel.76ee4591.webp | 102.94kb | +0.00b | 165.00ms | +17.00ms ⚠️ |
| dist/legacy/index.css | 29.00b | +0.00b | 1.38s | +78.00ms ⚠️ |

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/modern/parcel.76ee4591.webp | 102.94kb | +0.00b | 11.00ms | -1.00ms 🚀 |
| dist/legacy/index.js | 243.00b | +0.00b | 17.00ms | -1.00ms 🚀 |
| dist/modern/index.js | 215.00b | +0.00b | 12.00ms | -1.00ms 🚀 |

</p></details>

<details><summary>packages/benchmarks/react-hn ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 14.26s | +52.00ms |
| Cached | 9.87s | -32.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/main/index.js | 34.00kb | +0.00b | 14.00ms | -1.00ms 🚀 |

</p></details>

<details><summary>packages/benchmarks/ak-editor 🚨</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | **FAILED** | -0.00ms |
| Cached | **FAILED** | -0.00ms |

#### Cold Bundles

*No bundles found, this is probably a failed build...*
#### Cached Bundles

*No bundles found, this is probably a failed build...*
</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/251482018740699657)**

**Comment by mischnic** [comment]
I only get MOZILLA_PKIX_ERROR_SELF_SIGNED_CERT (which is expected) without this change.

Does setting that value to false instead if fix it as well? That is their documentation says: https://developer.mozilla.org/en-US/docs/Mozilla/Security/x509_Certificates#Generate_the_end_entity_certificate_2

**Comment by DeMoorJasper** [comment]
@mischnic well I couldn't really reproduce the issue, just thought it might be good to fix it as it won't negatively impact anyone.

I did however notice firefox being thrown of by the altNames not including localhost or host if one is defined and updated that in the certificate generation as well

File: packages/core/utils/src/generateCertificate.js [hunk 1]
```diff
@@ -63,12 +64,30 @@ export default async function generateCertificate(
...
+    },
+    {
+      type: 7, // IP
+      ip: '127.0.0.1',
+    },
+  ];
+
+  if (host) {
+    altNames.push({
+      type: 2, // DNS
```
Comment (DeMoorJasper) [inline packages/core/utils/src/generateCertificate.js]: Not entirely sure if we should check if host is an ip or domain here to update type accordingly. I think type 2 should work with both ip and domain.

**Pull request merged**